bitchat/Services/BluetoothMeshService.swift

```diff
@@ -2869,9 +2869,20 @@ class BluetoothMeshService: NSObject {
         case .noiseEncrypted:
             // Handle Noise encrypted message
             let senderID = packet.senderID.hexEncodedString()
-            if !isPeerIDOurs(senderID) {
+            if let recipientIDData = packet.recipientID,
+               isPeerIDOurs(recipientIDData.hexEncodedString())
+               && !isPeerIDOurs(senderID) {
```

Comment:
We (Alice) tried to decrypt every package, even if the package is from Bob to Carol. We then send a NACK to Bob, tearing down our intact noise session with Bob.

This should fix it but I strongly suggest to filter out for our packets in a centralized place instead of in each handler, and also decide to relay packets with `broadcastPacket(relayPacket)` in a centralized place too.

Some of the remaining handlers still have this issue.